llarp/crypto/types.cpp

```diff
@@ -32,14 +33,44 @@ namespace llarp
     return oxenc::to_hex(begin(), end());
   }
 
+  PubKey::operator RouterID() const
+  {
+    return {as_array()};
+  }
+
+  PubKey&
+  PubKey::operator=(const byte_t* ptr)
+  {
+    std::copy(ptr, ptr + SIZE, begin());
+    return *this;
+  }
+
+  bool
+  operator==(const PubKey& lhs, const PubKey& rhs)
+  {
+    return lhs.as_array() == rhs.as_array();
+  }
+
+  bool
+  operator==(const PubKey& lhs, const RouterID& rhs)
+  {
+    return lhs.as_array() == rhs.as_array();
+  }
+
+  bool
+  operator==(const RouterID& lhs, const PubKey& rhs)
+  {
+    return lhs.as_array() == rhs.as_array();
+  }
+
   bool
   SecretKey::LoadFromFile(const fs::path& fname)
```

Comment:
I feel like we should get rid of these `LoadFromFile` methods and just have a single function something like:
```C++
template <typename T, typename = std::enable_if_t<is_aligned_buffer<T>>>
T load_from_file(const fs::path& fname) {
    ...
}

// ...
auto sk = load_from_file<SecretKey>("filename");
```
Having these nearly-identical `LoadFromFile` scatter all over the codebase (making `fs.hpp` included almost everywhere) seems sub-optimal.

Reply:
I agree, but question and a note:

Q: What `LoadFromFile` methods are you referring to? I only see the one in `SecretKey`
```
dan@demeter:~/oxen/lokinet/llarp$ grep -R "LoadFromFile" *
config/key_manager.cpp:    return key.LoadFromFile(path);
crypto/types.hpp:    LoadFromFile(const fs::path& fname);
crypto/types.cpp:  SecretKey::LoadFromFile(const fs::path& fname)
exit/session.cpp:    return exit_key.LoadFromFile(fname);
```

Note: We already have other free methods for reading files directly to buffers or string [here](https://github.com/oxen-io/lokinet/blob/07271f9ae7a9d0f39c3b866d06d625e218315419/llarp/util/file.hpp#L13). Their former name is absolutely terrible so I re-named them, but maybe we can modify these to be more what you're thinking of?